### PR TITLE
Laboratory work #3, Maksimova Ann - 19FPL2

### PR DESCRIPTION
**Pull request by amksmva**

### CODE REVIEW

File: lab_3/main.py [hunk 1]
```diff
@@ -17,38 +17,89 @@ def tokenize_by_sentence(text: str) -> tuple:
...
+        return ()
+
+    letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u',
+               'v', 'w', 'x', 'y', 'z']
+
+    for symbol in text.lower():
+        if symbol in letters:
+            break
+    else:
+        return ()
```
Comment (WhiteJaeger) [inline lab_3/main.py]: It is both time and memory consuming - try to find another way to check for correct input.

File: lab_3/main.py [hunk 2]
```diff
@@ -17,38 +17,89 @@ def tokenize_by_sentence(text: str) -> tuple:
...
+
+    text = text.replace('!', '.')
+    text = text.replace('?', '.')
```
Comment (WhiteJaeger) [inline lab_3/main.py]: Can be done in one line.

File: lab_3/main.py [hunk 3]
```diff
@@ -17,38 +17,89 @@ def tokenize_by_sentence(text: str) -> tuple:
...
+        return self.storage[letter]
 
     def update(self, corpus: tuple) -> int:
         """
         Fills a storage by letters from the corpus
         :param corpus: a tuple of sentences
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(corpus, tuple) or corpus is None:
```
Comment (WhiteJaeger) [inline lab_3/main.py]: `corpus is None` can be shortened.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_3/main.py [hunk 4]
```diff
@@ -17,38 +20,92 @@ def tokenize_by_sentence(text: str) -> tuple:
...
-    pass
+    if not isinstance(text, str) or not text:
+        return ()
+
+    letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u',
+               'v', 'w', 'x', 'y', 'z']
+
+    check_appropriate_input = False
+    for symbol in text.lower():
+        if symbol in letters:
```
Comment (WhiteJaeger) [inline lab_3/main.py]: You check that at least one symbol in text is a letter?

**Review by WhiteJaeger — APPROVED** [review]
Good job! You are done here. Please, explain or re-write the commented part.

File: lab_3/main.py (hunk 4 above)
Comment (amksmva) [reply lab_3/main.py]: Yes!!

**Pull request closed without merging**